### PR TITLE
Add option to set the noise model when using the IonQ-hosted simulator

### PR DESCRIPTION
**Pull request by 1tnguyen**
### Description

Trying the IonQ simulator target, I saw that we didn't have an option to use the free hardware noise model simulation hence adding it in this PR.

- Add a CLI option (`--ionq-noise-model`) for users to set the noise model (ideal/aria-1/harmony).

- Send on to the job request JSON according to https://ionq.com/docs/get-started-with-hardware-noise-model-simulation

- Update the example to include instructions on how to use noise model simulation.

### CODE REVIEW

**Comment by 1tnguyen** [comment]
@splch It seems like I cannot add you as a reviewer, but I'd appreciate it if you could have a quick look :-)

File: runtime/cudaq/platform/default/rest/helpers/ionq/IonQServerHelper.cpp [hunk 1]
```diff
@@ -123,6 +126,18 @@ IonQServerHelper::createJob(std::vector<KernelExecution> &circuitCodes) {
...
     job["target"] = backendConfig.at("target");
+    // Add noise model config to the JSON job request
+    if (keyExists("noise_model")) {
+      if (backendConfig.at("target") == "simulator") {
+        nlohmann::json noiseModel;
+        noiseModel["model"] = backendConfig.at("noise_model");
+        job["noise"] = noiseModel;
+      } else {
+        printf("IonQ noise model setting is only valid for the 'simulator' "
+               "ionq-machine target. This will be ignored.\n");
```
Comment (boschmitt) [inline runtime/cudaq/platform/default/rest/helpers/ionq/IonQServerHelper.cpp:137]: We should silently ignore this argument. Maybe use `cudaq::info` to output this message when we run a program  with the appropriate log level. The reasoning here is that this option will be given to the compiler, but the message will only be shown when running a compiled program.

So a user would compile using `nvq++ ... --ionq-noise-model <model>`, get not warnings/errors, but then see this massage when running the produced output `./a.out`. We need to make the driver/compiler validate these things

Comment (1tnguyen) [reply runtime/cudaq/platform/default/rest/helpers/ionq/IonQServerHelper.cpp]: Thanks @boschmitt. That's a good point. I've updated the PR to perform this validation at compile time as suggested.

**Review by amccaskey — COMMENTED** [review]
Overall looks good to me. Is this something we should add to later to the IonQ nightly tester @bettinaheim @boschmitt

**Pull request merged**